tests_pytest/common_tests/unit_tests/test_model_collector.py

```diff
@@ -272,3 +352,113 @@ def test_update_statistics_called(self, fw_impl_mock, fw_info_mock):
+        self.node3.is_activation_quantization_enabled = Mock(return_value=True)
+        self.node3.is_weights_quantization_enabled = Mock(return_value=False)
+        self.node3.is_fln_quantization = Mock(return_value=False)
+        
+        self.node4 = build_node('node4', output_shape=input_shape, qcs=[build_qc(4), build_qc(2)])
+        self.node4.is_activation_quantization_enabled = Mock(return_value=False)
+        self.node4.is_weights_quantization_enabled = Mock(return_value=False)
+        self.node4.is_fln_quantization = Mock(return_value=True)
+
+        mock_nodes_list = [self.node1, self.node2, self.node3, self.node4]
+        for node in mock_nodes_list:
+            node.is_quantization_preserving = Mock(return_value=False)
+
+        self.graph = Graph('g',
+                           input_nodes=[self.node1],
+                           nodes=mock_nodes_list,
+                           output_nodes=[OutTensor(self.node3, 0)],
+                           edge_list=[Edge(self.node1, self.node2, 0, 0), Edge(self.node2, self.node3, 0, 0)])
+
+        # Simulate model inference outputs.
+        self.fake_output1 = np.random.randn(*input_shape)
+        self.fake_output2 = np.random.randn(*input_shape)
+        self.fake_output3 = np.random.randn(*input_shape)
+        fw_impl_mock.run_model_inference.return_value = [self.fake_output1, self.fake_output2, self.fake_output3]
+        fw_impl_mock.to_numpy.side_effect = lambda x: x
+
+        fw_info_mock.get_kernel_op_attributes.return_value = [None]
+
+        # Set up a fake Hessian service with a predetermined hessian result.
+        self.fake_hessian_info_service = Mock(spec_set=HessianInfoService)
+        self.fake_hessian_result = np.random.randn(*input_shape)
+        self.fake_hessian_info_service.fetch_hessian = Mock(
+            wraps=self.fake_hessian_info_service.fetch_hessian,
+            return_value={'node1': self.fake_hessian_result, 'node2': self.fake_hessian_result}
+        )
+
+        self.qc = DEFAULTCONFIG
+        self.infer_input = [np.random.randn(*input_shape)]
+
+    def test_infer_without_hessian_in_FLN(self, fw_impl_mock, fw_info_mock):
+        """
+        In FLN node,verify that ModelCollector.infer calls run_model_inference without fetching hessian data
+        when activation_error_method is not HMSE.
+        """
+        self.qc.activation_error_method = QuantizationErrorMethod.MSE
+        mc = ModelCollector(self.graph, fw_impl_mock, fw_info_mock, qc=self.qc, hessian_info_service=self.fake_hessian_info_service)
+
+        mc.infer(self.infer_input)
+        # Check that inference runs without requiring gradients.
+        fw_impl_mock.run_model_inference.assert_called_once_with(mc.model, self.infer_input, requires_grad=False)
+        # Confirm that the Hessian service is not used.
+        mc.hessian_service.fetch_hessian.assert_not_called()
+
+    def test_infer_with_hessian_in_FLN(self, fw_impl_mock, fw_info_mock):
```

Comment:
There is no need to check FLN with hessian for stats collection.
In fact, I think we should forbid this in the code, and make FLN statistics to never use hessians for statistic collection (I'll add a comment about it separately).
So - you can modify the tests to remove the fake Hessain Service from the setup, and keep only the test without hessians.